admiral/debug_functions.sqf

```diff
@@ -287,6 +287,88 @@ adm_debug_fnc_updateLineMarker = {
     DEBUG("admiral.debug",FMT_3("Updated line marker '%1' position between '%2' and '%3'.",_markerName,_posFrom,_posTo));
 };
 
+adm_debug_fnc_createDebugCounterMarker = {
+    FUN_ARGS_3(_side,_type,_xPos);
+    
+    PVT_2(_marker,_markerType);
+    _markerType = if (_type != "total") then {
+        _type;
+    } else {
+        COUNTER_DEBUG_MARKER_TOTAL;
+    };
+    _marker = [format ["adm_counter_%1_%2", _side,_type], [_xPos, 50, 0], "ICON", _markerType, [_side] call adm_debug_fnc_getSideColor, COUNTER_DEBUG_MARKER_SIZE] call adm_common_fnc_createLocalMarker;
+    _marker setMarkerTextLocal "0";
+    
+    _marker;
+};
+
+adm_debug_fnc_createDebugFactionCounterMarkers = {
+    FUN_ARGS_2(_side,_xPos);
+
+    DECLARE(_debugMarkers) = [];
+    PVT_1(_marker);
+    {
+        _marker = [_side, _x, _xPos] call adm_debug_fnc_createDebugCounterMarker;
+        _debugMarkers pushBack _marker;
+        _xPos = _xPos + COUNTER_DEBUG_MARKER_X_INCREMENT;
+    } foreach GROUP_TYPE_DEBUG_MARKERS;
+    _marker = [_side, "total", _xPos] call adm_debug_fnc_createDebugCounterMarker;
+    _debugMarkers pushBack _marker;
+    _xPos = _xPos + COUNTER_DEBUG_MARKER_X_INCREMENT;
```

Comment:
Still, mutating a parameter is a bad idea. Either make a local copy (_currentXPos) or calculate the value. I know that in this case it doesn't matter, but when you are passing a reference it can cause problems, that are really hard to debug.